src/EditorFeatures/Core/Implementation/Classification/SemanticClassificationTaggerProvider.cs

```diff
@@ -38,45 +56,135 @@ internal partial class SemanticClassificationTaggerProvider : AbstractAsynchrono
             _typeMap = typeMap;
         }
 
-        // We don't want to remove a tag just because it intersected an edit.  This can 
-        // cause flashing when a edit touches the edge of a classified symbol without
-        // changing it.  For example, if you have "Console." and you remove the <dot>,
-        // then you don't want to remove the classification for 'Console'.
-        protected override bool RemoveTagsThatIntersectEdits => false;
-
-        protected override SpanTrackingMode SpanTrackingMode => SpanTrackingMode.EdgeExclusive;
+        public override ITaggerEventSource CreateEventSource(ITextView textViewOpt, ITextBuffer subjectBuffer)
+        {
+            // Note: we don't listen for OnTextChanged.  Text changes to this this buffer will get
+            // reported by OnSemanticChanged.
+            return TaggerEventSources.Compose(
+                TaggerEventSources.OnSemanticChanged(subjectBuffer, TaggerDelay.Short, _semanticChangeNotificationService),
+                TaggerEventSources.OnDocumentActiveContextChanged(subjectBuffer, TaggerDelay.Short));
```

Comment:
Shouldn't anybody asking for changes in semantics to a buffer get OnDocumentActiveContextChanged for free? Is there ever a case where you want one but not the other?